### PR TITLE
add initial zlib module with crc32

### PR DESCRIPTION
**Pull request by anonrig**
Initializes `node:zlib` and implements only `crc32` for now.

### CODE REVIEW

File: src/workerd/api/node/zlib-util.c++ [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include "zlib-util.h"
+
+#include "zlib.h"
+
+namespace workerd::api::node {
+
+namespace {
+// This function is required to avoid function name clash.
+uint32_t crc32Impl(uint32_t value, unsigned char* bytes, size_t size) {
+  return crc32(value, reinterpret_cast<const Bytef*>(bytes), size);
```
Comment (jasnell) [inline src/workerd/api/node/zlib-util.c++:13]: nit: a comment here that Bytef is from zlib would be helpful I think. It's a bit obscure

File: src/workerd/api/node/zlib-util.c++ [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}
+} // namespace
+
+uint32_t ZlibUtil::crc32(kj::Array<kj::byte> data, uint32_t value) {
+  return crc32Impl(value, data.asBytes().begin(), data.size());
```
Comment (jasnell) [inline src/workerd/api/node/zlib-util.c++:18]: nit: The "asBytes()" here is not necessary

**Review by jasnell — APPROVED** [review]

File: src/workerd/api/node/zlib-util.c++ (hunk 2 above)
Comment (anonrig) [reply src/workerd/api/node/zlib-util.c++]: Thanks. Updating it now!

File: src/workerd/api/node/tests/zlib-nodejs-test.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
+import {
+  strictEqual,
+  throws,
+} from 'node:assert';
+import { Buffer } from 'node:buffer';
+import { crc32 } from 'node:zlib';
+
+
+// The following test data comes from
+// https://github.com/zlib-ng/zlib-ng/blob/5401b24/test/test_crc32.cc
```
Comment (garrettgu10) [inline src/workerd/api/node/tests/zlib-nodejs-test.js:10]: love the discussion on testcase sourcing, hope to see more of this in the future

**Review by garrettgu10 — APPROVED** [review]
Nice!

**Pull request merged**